DB_HW_climbs_for_a_mountaineering_club

alisherr003 opened the pull request.

alisherr003 committed:
Add files via upload

Marip7 reviewed (changes requested):
Hi Alisher,
Please see the red comments in the attached docx file. [DB_Alisher_Sulaymanov_HW_climbs_for_a_mountaineering_club_description.docx](https://github.com/user-attachments/files/19341216/DB_Alisher_Sulaymanov_HW_climbs_for_a_mountaineering_club_description.docx)
You need to completely redesign the relationships between the tables in the diagram and correctly specify the PK and FK. An example of how to represent a many-to-many relationship is attached. 
![example](https://github.com/user-attachments/assets/6b115ed1-e9ad-4c46-ba51-36e66b479a6d)

Next, according to the assignment, you need to add the constraints: KEYs, CHECK, NOT NULL, UNIQUE, and DEFAULT.

alisherr003 commented:
Dear Marina
I hope you are well.

Thank you for quick checking my assignments and replying.
I corrected all things on Database model. But i can't open the attached doc file, it returns me the "404 This is not the web page  you are looking for". Please help me on this situation.

alisherr003 commented:
Dear Marina
I hope you are well.

Thank you for quick checking my assignments and replying.
I corrected all things on Database model. But i can't open the attached doc
file, it returns me the "404 This is not the web page you are looking for".
Please help me on this situation.

ср, 19 мар. 2025 г. в 18:14, Marip7 ***@***.***>:

> ***@***.**** requested changes on this pull request.
>
> Hi Alisher,
> Please see the red comments in the attached docx file.
> DB_Alisher_Sulaymanov_HW_climbs_for_a_mountaineering_club_description.docx
> <https://github.com/user-attachments/files/19341216/DB_Alisher_Sulaymanov_HW_climbs_for_a_mountaineering_club_description.docx>
> You need to completely redesign the relationships between the tables in
> the diagram and correctly specify the PK and FK. An example of how to
> represent a many-to-many relationship is attached.
> example.png (view on web)
> <https://github.com/user-attachments/assets/6b115ed1-e9ad-4c46-ba51-36e66b479a6d>
>
> Next, according to the assignment, you need to add the constraints: KEYs,
> CHECK, NOT NULL, UNIQUE, and DEFAULT.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#pullrequestreview-2698391211>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BD4SRVZYXV5RY7WVGVX4D4L2VFUU3AVCNFSM6AAAAABZG2YPD6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMOJYGM4TCMRRGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Marip7 committed:
Added comments in the file

Marip7 commented:
Hi Alisher, 
I've committed this file directly in your branch (here is the direct link https://github.com/alisherr003/Alisher_Sulaymanov_SP25/blob/DB_HW_climbs_for_a_mountaineering_club/DB_Alisher_Sulaymanov_HW_climbs_for_a_mountaineering_club_description.docx )
Please correct all errors ASAP.

alisherr003 commented:
*Good evening, Marina,*

I hope you're doing well. *Sorry for the late reply.*

I have booked the meeting for *tomorrow from 18:40 to 19:10 (UTC+05).* However,
I have a small request—would it be possible to reschedule it for *19:40 –
20:10 (UTC+05) or later?*

I will be working until *18:30*, and I may not be able to make it on time.

Would this be possible for you?

*Thank you in advance!*
Best regards

Alisher Sulaymanov


ср, 19 мар. 2025 г. в 18:14, Marip7 ***@***.***>:

> ***@***.**** requested changes on this pull request.
>
> Hi Alisher,
> Please see the red comments in the attached docx file.
> DB_Alisher_Sulaymanov_HW_climbs_for_a_mountaineering_club_description.docx
> <https://github.com/user-attachments/files/19341216/DB_Alisher_Sulaymanov_HW_climbs_for_a_mountaineering_club_description.docx>
> You need to completely redesign the relationships between the tables in
> the diagram and correctly specify the PK and FK. An example of how to
> represent a many-to-many relationship is attached.
> example.png (view on web)
> <https://github.com/user-attachments/assets/6b115ed1-e9ad-4c46-ba51-36e66b479a6d>
>
> Next, according to the assignment, you need to add the constraints: KEYs,
> CHECK, NOT NULL, UNIQUE, and DEFAULT.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#pullrequestreview-2698391211>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BD4SRVZYXV5RY7WVGVX4D4L2VFUU3AVCNFSM6AAAAABZG2YPD6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMOJYGM4TCMRRGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Marip7 commented:
Hi Alisher, 
Please don’t worry. 
I rescheduled the screening from 18:40 to 20:30.

alisherr003 commented:
Dear Marina

Thank you for understanding)
See you.

чт, 3 апр. 2025 г. в 05:22, Marip7 ***@***.***>:

> Hi Alisher,
> Please don’t worry.
> I rescheduled the screening from 18:40 to 20:30.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2774032714>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BD4SRV6XGFSHEJL4KPUZDHL2XR5LHAVCNFSM6AAAAABZG2YPD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZUGAZTENZRGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: Marip7]*Marip7* left a comment
> (alisherr003/Alisher_Sulaymanov_SP25#3)
> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2774032714>
>
> Hi Alisher,
> Please don’t worry.
> I rescheduled the screening from 18:40 to 20:30.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2774032714>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BD4SRV6XGFSHEJL4KPUZDHL2XR5LHAVCNFSM6AAAAABZG2YPD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZUGAZTENZRGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

alisherr003 commented:
Dear Marina

How are you doing?

I am here in meeting )

чт, 3 апр. 2025 г. в 07:46, Alisher Sulaymanov <
***@***.***>:

> Dear Marina
>
> Thank you for understanding)
> See you.
>
> чт, 3 апр. 2025 г. в 05:22, Marip7 ***@***.***>:
>
>> Hi Alisher,
>> Please don’t worry.
>> I rescheduled the screening from 18:40 to 20:30.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2774032714>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BD4SRV6XGFSHEJL4KPUZDHL2XR5LHAVCNFSM6AAAAABZG2YPD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZUGAZTENZRGQ>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>> [image: Marip7]*Marip7* left a comment
>> (alisherr003/Alisher_Sulaymanov_SP25#3)
>> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2774032714>
>>
>> Hi Alisher,
>> Please don’t worry.
>> I rescheduled the screening from 18:40 to 20:30.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2774032714>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BD4SRV6XGFSHEJL4KPUZDHL2XR5LHAVCNFSM6AAAAABZG2YPD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZUGAZTENZRGQ>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

alisherr003 commented:
*Sorry, i meant that i will wait you on 20:30 (UTC +05) *

чт, 3 апр. 2025 г. в 19:39, Alisher Sulaymanov <
***@***.***>:

> Dear Marina
>
> How are you doing?
>
> I am here in meeting )
>
> чт, 3 апр. 2025 г. в 07:46, Alisher Sulaymanov <
> ***@***.***>:
>
>> Dear Marina
>>
>> Thank you for understanding)
>> See you.
>>
>> чт, 3 апр. 2025 г. в 05:22, Marip7 ***@***.***>:
>>
>>> Hi Alisher,
>>> Please don’t worry.
>>> I rescheduled the screening from 18:40 to 20:30.
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2774032714>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/BD4SRV6XGFSHEJL4KPUZDHL2XR5LHAVCNFSM6AAAAABZG2YPD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZUGAZTENZRGQ>
>>> .
>>> You are receiving this because you authored the thread.Message ID:
>>> ***@***.***>
>>> [image: Marip7]*Marip7* left a comment
>>> (alisherr003/Alisher_Sulaymanov_SP25#3)
>>> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2774032714>
>>>
>>> Hi Alisher,
>>> Please don’t worry.
>>> I rescheduled the screening from 18:40 to 20:30.
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2774032714>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/BD4SRV6XGFSHEJL4KPUZDHL2XR5LHAVCNFSM6AAAAABZG2YPD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZUGAZTENZRGQ>
>>> .
>>> You are receiving this because you authored the thread.Message ID:
>>> ***@***.***>
>>>
>>

alisherr003 commented:
Dear Marina

i am waiting for you ))

чт, 3 апр. 2025 г. в 19:40, Alisher Sulaymanov <
***@***.***>:

> *Sorry, i meant that i will wait you on 20:30 (UTC +05) *
>
> чт, 3 апр. 2025 г. в 19:39, Alisher Sulaymanov <
> ***@***.***>:
>
>> Dear Marina
>>
>> How are you doing?
>>
>> I am here in meeting )
>>
>> чт, 3 апр. 2025 г. в 07:46, Alisher Sulaymanov <
>> ***@***.***>:
>>
>>> Dear Marina
>>>
>>> Thank you for understanding)
>>> See you.
>>>
>>> чт, 3 апр. 2025 г. в 05:22, Marip7 ***@***.***>:
>>>
>>>> Hi Alisher,
>>>> Please don’t worry.
>>>> I rescheduled the screening from 18:40 to 20:30.
>>>>
>>>> —
>>>> Reply to this email directly, view it on GitHub
>>>> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2774032714>,
>>>> or unsubscribe
>>>> <https://github.com/notifications/unsubscribe-auth/BD4SRV6XGFSHEJL4KPUZDHL2XR5LHAVCNFSM6AAAAABZG2YPD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZUGAZTENZRGQ>
>>>> .
>>>> You are receiving this because you authored the thread.Message ID:
>>>> ***@***.***>
>>>> [image: Marip7]*Marip7* left a comment
>>>> (alisherr003/Alisher_Sulaymanov_SP25#3)
>>>> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2774032714>
>>>>
>>>> Hi Alisher,
>>>> Please don’t worry.
>>>> I rescheduled the screening from 18:40 to 20:30.
>>>>
>>>> —
>>>> Reply to this email directly, view it on GitHub
>>>> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2774032714>,
>>>> or unsubscribe
>>>> <https://github.com/notifications/unsubscribe-auth/BD4SRV6XGFSHEJL4KPUZDHL2XR5LHAVCNFSM6AAAAABZG2YPD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZUGAZTENZRGQ>
>>>> .
>>>> You are receiving this because you authored the thread.Message ID:
>>>> ***@***.***>
>>>>
>>>

Marip7 commented:
I'm running late with the previous student. Let's start 30 minutes later, at 21:00 your time.

Marip7 commented:
I wrote you in Teams, did you read it?

Marip7 commented:
New link will be:
https://teams.microsoft.com/l/meetup-join/19%3ameeting_NjBlNGU4NzgtZmQwNC00NmZmLTlkMTEtNzQwZmQyMDYwMWZk%40thread.v2/0?context=%7b%22Tid%22%3a%22b41b72d0-4e9f-4c26-8a69-f949f367c91d%22%2c%22Oid%22%3a%22e8abf619-4c6a-43cc-a25e-303847430b15%22%7d

alisherr003 commented:
Dear Marina

It is just ok, don't worry. I will wait you )

чт, 3 апр. 2025 г. в 20:45, Marip7 ***@***.***>:

> New link will be:
>
> https://teams.microsoft.com/l/meetup-join/19%3ameeting_NjBlNGU4NzgtZmQwNC00NmZmLTlkMTEtNzQwZmQyMDYwMWZk%40thread.v2/0?context=%7b%22Tid%22%3a%22b41b72d0-4e9f-4c26-8a69-f949f367c91d%22%2c%22Oid%22%3a%22e8abf619-4c6a-43cc-a25e-303847430b15%22%7d
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2776235189>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BD4SRV7CVLI67UISJMEHCRL2XVJSXAVCNFSM6AAAAABZG2YPD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZWGIZTKMJYHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: Marip7]*Marip7* left a comment
> (alisherr003/Alisher_Sulaymanov_SP25#3)
> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2776235189>
>
> New link will be:
>
> https://teams.microsoft.com/l/meetup-join/19%3ameeting_NjBlNGU4NzgtZmQwNC00NmZmLTlkMTEtNzQwZmQyMDYwMWZk%40thread.v2/0?context=%7b%22Tid%22%3a%22b41b72d0-4e9f-4c26-8a69-f949f367c91d%22%2c%22Oid%22%3a%22e8abf619-4c6a-43cc-a25e-303847430b15%22%7d
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/alisherr003/Alisher_Sulaymanov_SP25/pull/3#issuecomment-2776235189>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BD4SRV7CVLI67UISJMEHCRL2XVJSXAVCNFSM6AAAAABZG2YPD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZWGIZTKMJYHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

alisherr003 committed:
Add files via upload

Hi Dear Marina, here is attached my HW 1 from relational database schema modeling. 
Please check it and let me to know)

Marip7 commented:
Hi Alisher, 
Sorry, I can't understand the relationships between the tables. They need to be drawn in a clear and understandable way, similar to this example https://global.discourse-cdn.com/sitepoint/original/3X/e/a/ea94fbd4e1e8730455bb4fa266a22f423065b43a.png
![image](https://github.com/user-attachments/assets/03944bed-f419-4692-bd95-6ccee92972f2)
In the climb_mountains table, climb_id should be part of the Primary Key (PK)

alisherr003 committed:
Add files via upload

Good evening Marina

Thank you for your comment, here is attached changed version of schema.
I tried as well as possible ))
If there is any mistake, please let me to know.

Marip7 commented:
Hi Alisher, 
Do you see my images?
Why are there two relationships between the tables? Please create **only one** relationship, and correctly link the tables that represent many-to-many relationships.
Here’s an example of how to fix it for three of your tables — **please do the same for the rest**:

![image](https://github.com/user-attachments/assets/98a82dd2-318d-417a-9030-2f4c52ed73e8)

Repeating once again:
In the climb_mountains and climber_climbs tables, **climb_id should be part of the Primary Key (PK)**.
Also, the file should be committed as a replacement with the same name, not added as a new file with a different name.
**Please delete all the old files**.
